test/org/sagebionetworks/bridge/services/SurveyServiceMockTest.java

```diff
@@ -120,7 +129,35 @@ public void successfulDeletePermanently() {
         verify(mockSurveyDao).deleteSurveyPermanently(keysCaptor.capture());
         assertEquals(survey, keysCaptor.getValue());
     }
-    
+
+    @Test(expected = BadRequestException.class)
+    public void logicallyDeleteSurveyNotEmptySharedModules() {
+        when(mockSharedModuleMetadataService.queryAllMetadata(anyBoolean(), anyBoolean(), anyString(), anySetOf(String.class)))
+                .thenReturn(ImmutableList.of(makeValidMetadata()));
```

Comment:
It would be worthwhile to verify the args to test that we're constructing our query properly.

Similarly in physicallyDeleteSurveyNotEmptySharedModules(), UploadSchemaServiceTest.deleteByIdNotEmptySharedModules() and UploadSchemaServiceTest.deleteByIdAndRevNotEmptySharedModules().